dss/events/handlers/index.py

```diff
@@ -218,9 +221,21 @@ def notify(subscription_id: str, subscription: dict, bundle_id: str, logger):
             "bundle_version": bundle_version
         }
     }
-    # TODO (mbaumann) Ensure webhooks are only delivered over verified HTTPS (unless maybe when running a test)
     callback_url = subscription['callback_url']
-    response = requests.post(callback_url, json=payload)
+
+    assert urlparse(callback_url).scheme in {"http", "https"}
+    if os.environ["DSS_DEPLOYMENT_STAGE"] not in {"dev", "staging"}:
+        hostname = urlparse(callback_url).hostname
+        for family, socktype, proto, canonname, sockaddr in socket.getaddrinfo(hostname, port=None):
+            assert ipaddress.ip_address(sockaddr[0]).is_global  # type: ignore
```

Comment:
What is this line checking for?

Reply:
It is checking that the webhook subscription isn't trying to make us send a notification to a local network address (not publicly routable, like https://127.0.0.1).

Reply:
Are there any AWS resources that authenticate based solely on source ip?

Reply:
Not that I know of, but I think it should be pretty unambiguously not allowed to try to send webhook deliveries to host-local or local-subnet endpoints in prod. I haven't found an effective attack vector but it seems quite dangerous. I would like to restrict this condition to prod only but since we don't have any official nomenclature for the prod deployment stages, I check if we're not in dev or staging. I suppose this breaks custom stages, so maybe I should change this to "prod" even though that's error prone if we ever deploy a production deployment that is not called "prod".

Reply:
Changed the stage check to compare against "prod".